crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -667,6 +667,51 @@ where
                     self.flow_merge(after_subject);
                 }
             }
+            ast::Stmt::Try(ast::StmtTry {
+                body,
+                handlers,
+                orelse,
+                finalbody,
+                is_star,
+                range: _,
+            }) => {
+                self.visit_body(body);
+
+                for except_handler in handlers {
+                    let ast::ExceptHandler::ExceptHandler(except_handler) = except_handler;
+                    let ast::ExceptHandlerExceptHandler {
+                        name: symbol_name,
+                        type_: handled_exceptions,
+                        body: handler_body,
+                        range: _,
+                    } = except_handler;
+
+                    if let Some(handled_exceptions) = handled_exceptions {
+                        self.visit_expr(handled_exceptions);
+                    }
+
+                    // If `handled_exceptions` above was `None`, it's something like `except as e:`,
+                    // which is invalid syntax. However, it's still pretty obvious here that the user
+                    // *wanted* `e` to be bound, so we should still create a definition here nonetheless.
+                    if let Some(symbol_name) = symbol_name {
+                        let symbol = self
+                            .add_or_update_symbol(symbol_name.id.clone(), SymbolFlags::IS_DEFINED);
+
+                        self.add_definition(symbol, {
+                            if *is_star {
+                                DefinitionNodeRef::ExceptStarHandler(except_handler)
+                            } else {
+                                DefinitionNodeRef::ExceptHandler(except_handler)
```

Comment:
I don't mind having different nodes, but I am interested in understanding your motivation for two variants over a boolean flag.

Reply:
There's not much in it. Carl and I thought it seemed marginally simpler to have two different variants rather than creating a dedicated struct with an `is_star: bool` flag inside the `ExceptHandler` variant. It also reinforces semantically that these are different kinds of definitions that create quite different types. But I'm very happy to do a dedicated struct if you prefer it!

Reply:
Nah that's fine. I was just interested in the reasoning and I'm sure if it's for the best if that was the conclusion you two came to.